### DISCUSSION TITLE
OpeMPTCProuter and Wireguard

### DISCUSSION THREAD

**Discussion by mansb123** [Ysurac/openmptcprouter#3033] [General]
Hi, my name is Filippo and i'm a master student of Polithecnic University of Marche (Italy). I'm using this project for thesis purpose and i've a problem. To better understand the problem i try to explain what i've configured on my lab:
-a VM running OpenMPTCProuter ext4 image downloaded from https://www.openmptcprouter.com/download
 -a VM where i've installed the VPS script.
-two Internet connections: Wi-Fi and 4G TIM SIM
Now, when i attacch the router to the VPS, the system tells me that 'MPTCP seems to be blocked' but i've solved the problem by enabling the 'MPTCP over VPN' function. With this function, the system works: i've done some tests for bandwidth and resilience. 
Now i'm sniffing the traffic with Wireshark because i want to see MPTCP packets but i can't see them because, as i think, they are encapsulated inside the Wireguard tunnel.
So My scope is now to decrypt this traffic and, searching the net, i've found a guide which explain the solution but something goes wrong with kernel-headers. It seems like there is a lack of this kernel-headers in the OpenWRTversion installed in the OpenMPTCProuter but i I just can't seem to install them.
Anyone can help me?

**Comment by Ysurac** [comment]
Hi,

You should use MPTCP over VPN only when MPTCP is blocked by a provider. You should really be sure it's the case (check VPS kernel) and not another problem because it's really rare in Europe to have MPTCP blocked.

kernels-headers are not installed on OpenMPTCProuter router side, compilation shouldn't be done on it: tools are not available for that.

**Reply by mansb123** [reply]
what i've to check in the VPS? if it can be useful i've sniffed the traffic not using MPTCP over VPN so i've left the error message MPTCP seems to be blocked and i see that the MPTCP packets are generated from my client and they are stopped by the Wi-Fi router which is one of the two out routers

**Reply by mansb123** [reply]
i've solved the problem. Thank you
